Tests/TensorFlowTests/OperatorTests/LinearAlgebraTests.swift

```diff
@@ -0,0 +1,67 @@
+// Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import XCTest
+@testable import TensorFlow
+
+final class LinearAlgebraTests: XCTestCase {
+    func testCholesky() {
+        let shapes = [[3, 3], [4, 2, 2], [2, 1, 16, 16]]
+        let permutations = [[1, 0], [0, 2, 1], [0, 1, 3, 2]] // To avoid permuting batch dimensions.
+        for (shape, permutation) in zip(shapes, permutations) {
+            let a = Tensor<Float>(randomNormal: TensorShape(shape))
+            let x = matmul(a, a.transposed(permutation: permutation)) // Make `a` positive-definite.
+            let l = cholesky(x)
+            let xReconstructed = matmul(l, l.transposed(permutation: permutation))
+            assertEqual(xReconstructed, x, accuracy: 1e-5)
+        }
+
+        // The expected value of the gradient was computed using the following Python code:
```

Comment:
@dan-zheng Do you think we should separate the test for cholesky function and its gradient? IMO unit testing is better when you test just for one thing. and if the cholesky function implementation undergoes any change in the future, such distinction can help locate any downstream errors.

Reply:
I think not separating is fine! I don't feel a strong need to have separate files or a massive file just for testing derivatives.